.ci_support/environment.yml

```diff
@@ -19,3 +19,4 @@ dependencies:
   - six >=1.14.0
   - sqlalchemy >=1.3.14
   - tqdm >=4.35.0
+  - xmltodict >=0.11.0
```

Comment:
Can you try using `defusedxml` because we have made good experience in parsing the VASP xml file. In addition we should add `pyyaml` if we want to support `yaml` import.

Reply:
I understand that `defusedxml` introduces security over `xml` library, but casting the content of the parser to a dictionary is not straight forward in both case. On the other hand `xmltodict`, [while it imports from `defusedexpat`](https://github.com/martinblech/xmltodict/blob/ae19c452ca000bf243bfc16274c060bf3bf7cf51/xmltodict.py#L5), make this casting really easy.

Reply:
Nevertheless, using `defusedxml`, the following function can also turn the content of the parsed xml file to a dictionary:
```
from defusedxml import ElementTree
from copy import copy

def dictify(r,root=True):
    if root:
        return {r.tag : dictify(r, False)}
    d=copy(r.attrib)
    if str(r.text).find('\n') == -1:
        d[r.tag]=r.text
    for x in r.findall("./*"):
        if len(x) >= 1:
            if x.tag not in d:
                d[x.tag]=[]
            d[x.tag].append(dictify(x,False))
        else:
            d[x.tag] = x.text
    return d

root = ElementTree.parse('sample.xml').getroot()
final_dict = dictify(root)
```

Reply:
@sudarsan-surendralal Can you comment on the choice of `defusedxml` vs `xmltodict` ? Maybe it is only related to the broken xml format in VASP.

Reply:
`defusedxml` is more secure for parsing xml files and was suggested by codeacy which complained when we used `xmltodict` earlier. We used it to parse `vasprun.xml` files for every single VASP calculation and it works pretty hassle-free. But yes, you need to tweak it a little bit to get the contents into a dictionary form

Reply:
@sudarsan-surendralal, I guess in the `xml` files for vasp, you have expected keys, and you look for some values. Am I right? But in our case, we need to cast the parsed xml into a dictionary or some sort of data containers to be saved as Datacontainer object.   
Another point, here, we have used `xmltodict` and so far codacy has not complained about it. It might be that `xmltodict` had been updated.

Reply:
> `vasprun`

@sudarsan-surendralal, I could not find the code where you parse `vasprun.xml` under `pyiron_atomistic/sphinx/`. Can you send me a link where `defusedxml` is used?

Reply:
> 
> 
> @sudarsan-surendralal, thanks for your response. I guess in the `xml` files for vasp, you have expected keys, and you look for some values. Am I right?

Yeah, that's true.

> But in our case, we need to cast the parsed xml into a dictionary or some sort of data containers to be saved as Datacontainer object.
> Another point, here, we have used `xmltodict` and so far Codacy has not complained about it. It might be that `xmltodict` has been updated.

Oh ok I was not aware of this. In that case maybe `xmltodict` is more appropriate for you. 

> @sudarsan-surendralal, I could not find the code where you parsed `vasprun.xml` under `pyiron_atomistic/sphinx/`. Can you send me a link where `defusedxml` is used?

It's in this file in the vasp directory https://github.com/pyiron/pyiron_atomistics/blob/master/pyiron_atomistics/vasp/vasprun.py